### PR TITLE
CSA-83

### PR DESCRIPTION
**Pull request by tymnik**
- adaptive for layout;
- changes in css according to new design;
- mobile menu for <1194px added;
- new icons added to sprite;
- text color changed to the whole project according to design;
- small bugfixes (throttle => debounce, navigation from logo etc.)

### CODE REVIEW

File: src/components/AuthNav/AuthNav.tsx [hunk 1]
```diff
@@ -6,24 +6,36 @@ import { buttonVariant } from 'components/ui/Button';
...
+    }
+  };
+
   return (
     <ul className={styles.authNavList}>
       <li>
         <Link
           to={ROUTES.LOGIN}
-          className={buttonVariant({ size: 'sm', variant: 'secondary' })}
+          className={`${buttonVariant({ size: 'sm', variant: 'secondary' })} ${isSmallScreen ? styles.mobBtn : ''}`}
```
Comment (VovaLyapota) [inline src/components/AuthNav/AuthNav.tsx]: use clsx instead of brackets

File: src/components/AuthNav/AuthNav.tsx [hunk 2]
```diff
@@ -6,24 +6,36 @@ import { buttonVariant } from 'components/ui/Button';
...
         </Link>
       </li>
       <li>
         <Link
           to={ROUTES.REGISTER}
-          className={buttonVariant({
-            size: 'sm',
-            variant: 'primary',
-          })}
+          className={`${buttonVariant({ size: 'sm', variant: 'primary' })} ${isSmallScreen ? styles.mobBtn : ''}`}
```
Comment (VovaLyapota) [inline src/components/AuthNav/AuthNav.tsx]: also clsx

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -1,8 +1,11 @@
 import { Link } from 'react-router-dom';
 
 import logo from 'src/assets/csa-logos/cluster-logo.svg';
+import feedbackIcon from 'src/assets/feedback.svg';
 import { ROUTES } from 'src/routing/routes';
 
+import Button from '../ui/Button';
```
Comment (VovaLyapota) [inline src/components/Footer/Footer.tsx]: use components/ui/Button, it`s more clear to understand

File: src/components/Footer/Footer.tsx [hunk 4]
```diff
@@ -1,8 +1,11 @@
...
+
 import Contacts from './Contacts/Contacts';
```
Comment (VovaLyapota) [inline src/components/Footer/Footer.tsx]: add index.ts to Contacts components, it mustn't repeat itself in path

File: src/components/Header/Header.tsx [hunk 5]
```diff
@@ -11,6 +12,7 @@ import Button, { buttonVariant } from 'components/ui/Button';
 import Icon from 'components/ui/Icon';
 import UserNav from 'components/UserNav';
 
+import MobMenu from './MobMenu/MobMenu';
```
Comment (VovaLyapota) [inline src/components/Header/Header.tsx]: index.ts in folder again

File: src/components/Header/Header.tsx [hunk 6]
```diff
@@ -37,56 +45,85 @@ const Header = () => {
     setIsNavVisible(true);
   };
 
+  const handleCloseMenu = () => {
```
Comment (VovaLyapota) [inline src/components/Header/Header.tsx]: I think you don't need a handleCloseMenu if you have a toggleMenu

File: src/components/Header/Header.tsx [hunk 7]
```diff
@@ -37,56 +45,85 @@ const Header = () => {
...
-    <header className={styles.header}>
-      <a href={ROUTES.HOME} rel="noopener noreferrer">
-        <img src={logo} alt="Career Skill Atlas logo" />
-      </a>
-      <div className={styles.headerMenu}>
-        {isSearchVisible ? (
-          <SearchBar
-            onClose={handleSearchClose}
-            className={isSearchVisible ? '' : styles.searchBarClosed}
+    <header className={clsx(isMenuActive && styles.sticky)}>
```
Comment (VovaLyapota) [inline src/components/Header/Header.tsx]: you can use a more clear syntax like clsx(styles.stiky: isMenuActive)

File: src/components/Header/Header.tsx [hunk 8]
```diff
@@ -37,56 +45,85 @@ const Header = () => {
...
-            </li>
-          </ul>
-        )}
-        {isAuthenticated ? <UserNav /> : <AuthNav />}
+        </Link>
+        <div className={styles.headerMenu}>
+          {isSearchVisible ? (
+            <SearchBar
+              onClose={handleSearchClose}
+              className={isSearchVisible ? '' : styles.searchBarClosed}
```
Comment (VovaLyapota) [inline src/components/Header/Header.tsx]: clsx

File: src/components/Header/MobMenu/MobMenu.tsx [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+import Icon from 'components/ui/Icon';
+
+import sprite from '../../../assets/sprite.svg';
+import { ROUTES } from '../../../routing/routes';
+import { selectIsLoggedIn } from '../../../store/user/selectors';
+import AuthNav from '../../AuthNav';
+import SearchBar from '../SearchBar/SearchBar';
+
+import UserNavMenu from './UserNavMenu/UserNavMenu';
+
```
Comment (VovaLyapota) [inline src/components/Header/MobMenu/MobMenu.tsx:16]: fix imports, You have to get rid of dots in paths and simplify your routs (../SearchBar/SearchBar => components/SearchBar)

File: src/components/Header/MobMenu/MobMenu.tsx [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+import styles from './MobMenu.module.css';
+
+interface MobMenuProps {
+  active: boolean;
+  toggleMenu: () => void;
+}
+
+const MobMenu = ({ active, toggleMenu }: MobMenuProps) => {
+  const [isSearchVisible, setIsSearchVisible] = useState(false);
+  const isAuthenticated = useSelector(selectIsLoggedIn);
```
Comment (VovaLyapota) [inline src/components/Header/MobMenu/MobMenu.tsx]: useAppSelector, use especial funcs for redux

File: src/components/Header/MobMenu/MobMenu.tsx [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+  };
+
+  const handleSearchClose = () => {
+    setIsSearchVisible(false);
+  };
+
+  return (
+    <div id="menu-overlay" className={styles.overlay}>
+      <div
+        className={clsx(styles.mobMenu, { [styles.active]: active })}
```
Comment (VovaLyapota) [inline src/components/Header/MobMenu/MobMenu.tsx]: why do you use square brackets here?

File: src/components/Header/MobMenu/MobMenu.tsx [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+          className={styles.mobMenuContent}
+          onClick={(e) => e.stopPropagation()}
+        >
+          <ul className={styles.mobMenuList}>
+            {menuItems.map((item, index) => (
+              <Link to={item.route} key={index}>
+                <li className={styles.mobMenuListItem} onClick={toggleMenu}>
+                  <svg className={styles.menuIcon}>
+                    <use xlinkHref={`${sprite}#${item.icon}`} />
+                  </svg>
```
Comment (VovaLyapota) [inline src/components/Header/MobMenu/MobMenu.tsx]: Use Icon components here

File: src/components/Header/MobMenu/MobMenu.tsx [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+                  {item.text}
+                </li>
+              </Link>
+            ))}
+            <li>
+              {isSearchVisible ? (
+                <SearchBar
+                  onClose={handleSearchClose}
+                  className={clsx(isSearchVisible && styles.searchBarClosed)}
```
Comment (VovaLyapota) [inline src/components/Header/MobMenu/MobMenu.tsx]: simplify clsx syntax

File: src/components/Header/MobMenu/MobMenu.tsx [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+                  className={styles.searchBarBtn}
+                  onClick={handleSearchClick}
+                  icon={<Icon id="search" className={styles.searchIcon} />}
+                >
+                  Search
+                </Button>
+              )}
+            </li>
+          </ul>
+          <div className={styles.decorativeLine}></div>
```
Comment (VovaLyapota) [inline src/components/Header/MobMenu/MobMenu.tsx]: i think you'd better to use single tag like <span/>

File: src/components/Header/MobMenu/UserNavMenu/UserNavMenu.tsx [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+import { useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import Icon from 'components/ui/Icon';
+
+import { deleteToken } from '../../../../api';
+import { useAppDispatch } from '../../../../hooks/redux';
+import { ROUTES } from '../../../../routing/routes';
+import { selectUser } from '../../../../store/user/selectors';
+import { logout } from '../../../../store/user/slice';
```
Comment (VovaLyapota) [inline src/components/Header/MobMenu/UserNavMenu/UserNavMenu.tsx]: fix imports

File: src/components/UserNav/UserNav.tsx [hunk 16]
```diff
@@ -38,6 +42,7 @@ const UserNav = () => {
     deleteToken();
     dispatch(logout());
     toggleDropdown();
+    onClose();
   };
```
Comment (VovaLyapota) [inline src/components/UserNav/UserNav.tsx:44]: I think we were talking about reducing this function

File: src/components/Footer/Footer.tsx [hunk 17]
```diff
@@ -26,8 +29,21 @@ const Footer = () => {
...
+            className={styles.feedbackButton}
+            variant="secondary"
+            size="lg"
+            icon={<img src={feedbackIcon} alt="Feedback" />}
+            iconPosition="before"
+          >
+            Leave us a feedback
+          </Button>
           <a href={'/'} target="_blank" rel="noopener noreferrer">
             <img
```
Comment (StanislavKharkiv) [inline src/components/Footer/Footer.tsx]: don't forget use constants for path

File: src/components/Header/Header.tsx [hunk 18]
```diff
@@ -20,6 +22,12 @@ const Header = () => {
   const location = useLocation();
   const [isSearchVisible, setIsSearchVisible] = useState(false);
   const [isNavVisible, setIsNavVisible] = useState(true);
+  const [isMenuActive, setIsMenuActive] = useState(false);
+
+  const toggleMenu = () => {
+    setIsMenuActive((prevState) => !prevState);
+    document.body.classList.toggle('menu-active', !isMenuActive);
+  };
```
Comment (StanislavKharkiv) [inline src/components/Header/Header.tsx]: Why go to DOM and change class in body? can we do it here in react component?

File: src/components/AuthNav/AuthNav.tsx [hunk 19]
```diff
@@ -6,24 +6,36 @@ import { buttonVariant } from 'components/ui/Button';
...
+
+const AuthNav = (props: AuthNavProps) => {
+  const { onClose } = props;
+  const isSmallScreen = window.innerWidth < 1194;
+
+  const handleButtonClick = () => {
+    if (onClose) {
+      onClose();
+    }
+  };
```
Comment (StanislavKharkiv) [inline src/components/AuthNav/AuthNav.tsx]: Why you call this props onClose. It's looks like onClick to menu item.
And you can just pass it in `<Link onClick={onClick} >`

File: src/components/AuthNav/AuthNav.tsx [hunk 20]
```diff
@@ -6,24 +6,36 @@ import { buttonVariant } from 'components/ui/Button';
...
 
-const AuthNav = () => {
+interface AuthNavProps {
+  onClose: () => void;
+}
+
+const AuthNav = (props: AuthNavProps) => {
+  const { onClose } = props;
+  const isSmallScreen = window.innerWidth < 1194;
+
```
Comment (StanislavKharkiv) [inline src/components/AuthNav/AuthNav.tsx]: I think better use media request inside css for this

File: src/components/Header/Header.module.css [hunk 21]
```diff
@@ -1,44 +1,86 @@
...
-  justify-content: space-between;
+  display: none;
+}
+
+@media screen and (min-width: 1194px) {
+  .headerMenu {
+    display: flex;
+    justify-content: space-between;
+  }
 }
```
Comment (StanislavKharkiv) [inline src/components/Header/Header.module.css]: Why you don't write in one @media all styles for one resolution ? Just combine all styles in one request

File: src/components/Header/Header.tsx [hunk 22]
```diff
@@ -37,56 +45,85 @@ const Header = () => {
...
+          {isSearchVisible ? (
+            <SearchBar
+              onClose={handleSearchClose}
+              className={isSearchVisible ? '' : styles.searchBarClosed}
+            />
+          ) : (
+            <ul
+              className={styles.headerNavMenu}
+              style={{ display: isNavVisible ? 'flex' : 'none' }}
+            >
```
Comment (StanislavKharkiv) [inline src/components/Header/Header.tsx]: I prefer avoid inline style

**Review by StanislavKharkiv — APPROVED** [review]

**Pull request merged**